packages/workbox-sw/index.mjs

```diff
@@ -0,0 +1,48 @@
+/**
+ * This module is a single import that can be used to dynamically import
+ * additional Workbox modules with no effort.
+ *
+ * @module workbox-sw
+ */
+
+/**
+ * This class can be used to make it easy to use the various parts of
+ * Workbox.
+ */
+class WorkboxSW {
+  /**
+   * You can specify whether to treat as a debug
+   * build and whether to use a CDN or a specific path when importing other
+   * workbox-modules
+   *
+   * @param {Object} options
+   */
+  constructor(options) {
+    this._options = Object.assign({
+      debug: self.location.hostname === 'localhost',
+      pathPrefix: null,
+    }, options);
+  }
+
+  /**
+   * Force a service worker to become active, instead of waiting. This is
+   * normally used in conjunction with `clientsClaim()`.
+   */
+  skipWaiting() {
```

Comment:
Couple of points re: `skipWaiting()` and `clientsClaim()`:

- In v2 they were booleans passed to the `WorkboxSW` constructor. I'm curious what's behind the change to making them methods.
- Neither of them need to be wrapped in `event.waitUntil()`, since they just return promises that resolve immediately.

Reply:
@jeffposnick main reason was that the config was largely pulled in from sw-precache config and I didn't see any valid in reason in parsing the config, then using the config to call an internal function that did this vs, just getting the dev to call the function.